tsdb/engine/tsm1/compact.go

```diff
@@ -573,7 +573,16 @@ func (c *Compactor) write(path string, iter KeyIterator) (err error) {
 		}
 
 		// Write the key and value
-		if err := w.WriteBlock(key, minTime, maxTime, block); err != nil {
+		err = w.WriteBlock(key, minTime, maxTime, block)
+		if err == ErrMaxBlocksExceeded {
+			if err := w.WriteIndex(); err != nil {
+				return err
+			}
+
+			return ErrMaxBlocksExceeded
```

Comment:
I don't mind the redundant return but I think specifying `ErrMaxBlocksExceeded` twice is redundant. You can just do a `return err` instead.